rocketpool-cli/update/update.go

```diff
@@ -0,0 +1,222 @@
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver/v4"
+	"github.com/urfave/cli"
+	"golang.org/x/crypto/openpgp"
+
+	"github.com/rocket-pool/smartnode/shared"
+	cliutils "github.com/rocket-pool/smartnode/shared/utils/cli"
+)
+
+// Settings
+const (
+	GithubAPIGetLatest string = "https://api.github.com/repos/rocket-pool/smartnode-install/releases/latest"
+	SigningKeyURL      string = "https://github.com/rocket-pool/smartnode-install/releases/download/v%s/smartnode-signing-key-v3.asc"
+	ReleaseBinaryURL   string = "https://github.com/rocket-pool/smartnode-install/releases/download/v%s/rocketpool-cli-%s-%s"
+)
+
+func getHttpClientWithTimeout() *http.Client {
+	return &http.Client{
+		Timeout: time.Second * 5,
+	}
+}
+
+func checkSignature(signatureUrl string, pubkeyUrl string, verification_target *os.File) error {
+	pubkeyResponse, err := http.Get(pubkeyUrl)
+	if err != nil {
+		return err
+	}
+	defer pubkeyResponse.Body.Close()
+	if pubkeyResponse.StatusCode != http.StatusOK {
+		return fmt.Errorf("public key request failed with code %d", pubkeyResponse.StatusCode)
+	}
+	keyring, err := openpgp.ReadArmoredKeyRing(pubkeyResponse.Body)
+	if err != nil {
+		return fmt.Errorf("error while reading public key: %w", err)
+	}
+
+	signatureResponse, err := http.Get(signatureUrl)
+	if err != nil {
+		return err
+	}
+	defer signatureResponse.Body.Close()
+	if signatureResponse.StatusCode != http.StatusOK {
+		return fmt.Errorf("signature request failed with code %d", signatureResponse.StatusCode)
+	}
+
+	entity, err := openpgp.CheckDetachedSignature(keyring, verification_target, signatureResponse.Body)
+	if err != nil {
+		return fmt.Errorf("error while verifying signature: %w", err)
+	}
+
+	for _, v := range entity.Identities {
+		fmt.Printf("Signature verified. Signed by: %s\n", v.Name)
+	}
+	return nil
+}
+
+// Update the Rocket Pool CLI
+func updateCLI(c *cli.Context) error {
```

Comment:
I see a few main sections to this function. I think it would be easier to see the main steps if they are split into their own functions. It would also help add context to the errors. I see the same error message repeated/similar to another one. What do you think?

Reply:
Yes, I think that's a fair comment. I've split out the initial version check, and the download/verify sections into their own functions, and I think `updateCLI` is pretty readable now :)